CMakeLists.txt

```diff
@@ -0,0 +1,423 @@
+#=============================================================================
+# Copyright 2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+cmake_minimum_required(VERSION 3.20.1 FATAL_ERROR)
```

Comment:
What is the reason to require minimum  version to be 3.20?

Reply:
I believe the reason is `3.20.1` is the first version with support for the `CUDA` language that doesn't require hacks or workarounds like reimplementing `FindCUDA`.

Reply:
OK, sounds good